repository/src/main/java/org/apache/atlas/repository/store/graph/v2/IndexRepairService.java

```diff
@@ -0,0 +1,234 @@
+package org.apache.atlas.repository.store.graph.v2;
+
+import org.apache.atlas.AtlasErrorCode;
+import org.apache.atlas.exception.AtlasBaseException;
+import org.apache.atlas.model.repair.*;
+import org.apache.atlas.repository.Constants;
+import org.apache.atlas.repository.graphdb.AtlasGraph;
+import org.apache.atlas.repository.graphdb.AtlasGraphQuery;
+import org.apache.atlas.repository.graphdb.AtlasVertex;
+import org.apache.atlas.repository.graphdb.janus.AtlasJanusGraph;
+import org.apache.commons.collections.CollectionUtils;
+import org.janusgraph.core.JanusGraph;
+import org.janusgraph.graphdb.database.util.StaleIndexRecordUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Service;
+
+import javax.inject.Inject;
+import java.util.*;
+
+@Service
+public class IndexRepairService {
+    private static final Logger LOG = LoggerFactory.getLogger(IndexRepairService.class);
+
+    // Index names
+    private static final String COMPOSITE_INDEX_QN_TYPE = "__u_qualifiedName__typeName";
+
+    private final AtlasGraph graph;
+
+    @Inject
+    public IndexRepairService(AtlasGraph graph) {
+        this.graph = graph;
+    }
+
+
+    /**
+     * Find corrupted vertex using single index (qualifiedName only)
+     */
+    public Optional<Long> findCorruptedSingleVertex(String qualifiedName) {
+        if (StringUtils.isBlank(qualifiedName)) {
+            return Optional.empty();
+        }
+
+        AtlasGraphQuery query = graph.query()
+                .has(Constants.UNIQUE_QUALIFIED_NAME, qualifiedName);
+
+        Iterator<AtlasVertex> vertices = query.vertices().iterator();
+
+        if (vertices.hasNext()) {
```

Comment:
Realized one thing: Please adda a failsafe if asset is not corrupted we will throw error. If important attrs are present we can't update. So we don't accidentally remove index record for an asset. It will be used by people outside of our team so this check i very important so we don't remove vertex/index from a valid asset.